docs/book/src/architecture/application-namespace.md

```diff
@@ -0,0 +1,11 @@
+## Application Namespace Reconciliation
+
+The Lifecycle of a application namespace includes the creation, updation and deletion of the primaza resources upon creation/deletion of the namespace.
+
+There is a Application Namespace Controller which does the reconciliation of application namespace and it runs on the Primaza main cluster.
+
+* Application Namespace Creation: When the application namespace is created, Service Bindings, Service Catalogs which are created into Primaza's namspace are forwarded to the matching application namepsaces of the workercluster.
+
+* Application Namespace Update: When a creation or update event on Cluster Environment occurs, its application namespaces are reconciled. The Service Catalogs and Service Bindings are pushed to each application Namespace.
+
+* Application Namespace Deletion: When the Cluster Environment is deleted, all the Primaza's resources (Service Catalog, Service Binding) are removed from the application namespaces.
```

Comment:
In the Update section you are talking about the ClusterEnvironment, but not in the other sections. I'd say we should definitely add more details about the relationship between Application/Service Namespaces and ClusterEnvironments. As an example we should say that ClusterEnvironments declare Application and Service Namespaces that are hosted in a Cluster for a given Environment. I'd propose to add that in the `namespace-reconciliation.md` file.

Finally, all events here should describe their relation with the ClusterEnvironment. In more details, I'd suggest something like the following:

```suggestion
* Application Namespace Creation: when an application namespace is added to the `applicationNamespaces` list of a ClusterEnvironment, the ClusterEnvironment controller checks the permissions in the target namespace and tries to push the Application Agent in it.
Finally, it will push in the namespace also the ServiceCatalog for the given environment and all the ServiceBinding generated from ServiceClaims declaring a `labelMatch` or that are targeting the namespace in the `applicationContext` property.
* Application Namespace Update: when an update event on the Application Namespace parent  ClusterEnvironment occurs, all application namespaces are reconciled as described in the `Application Namespace Creation` and `Application Namespace Deletion`.
* Application Namespace Deletion: when an Application Namespace is deleted from its parent ClusterEnvironment's `applicationNamespaces` list or when the ClusterEnvironment itself is deleted, the Application Agent is deleted from the target namespace.
As a result, this will trigger a deletion of all the Primaza's resources in the namespace.
This deletion is actually relying on kubernetes' ownership.
```


Applies also to `service-namespace.md`

Reply:
yes